### PR TITLE
Add command-line execution requirements

### PR DESCRIPTION
**Pull request by ian-r-rose**
cc @vahedi25411, opening this PR for discussion

### CODE REVIEW

File: automated_walk_bike_counter/gui/app.py [hunk 1]
```diff
@@ -30,6 +37,102 @@ def run(self):
...
+        tracker.color_table = color_table
+        tracker.input_camera_type = config.input_type
+        tracker.camera_id = config.camera_id
+        tracker.trackObjects(config)
+
+
 def main():
-    app = Application()
-    app.run()
+    console = True
```
Comment (ian-r-rose) [inline automated_walk_bike_counter/gui/app.py]: Let's add another CLI argument that determines whether to run in headless mode.

File: automated_walk_bike_counter/gui/app.py [hunk 2]
```diff
@@ -30,6 +37,102 @@ def run(self):
...
+        allowed_objects = config.VALID_OBJECTS
+        for i in range(len(allowed_objects)):
+            print(config.search_objects)
+            print(config.objects_colors)
+            print(allowed_objects[i])
+
+            if allowed_objects[i] in config.search_objects:
+                object_color = config.objects_colors[config.search_objects.index(allowed_objects[i])]
+                c = object_color.lstrip('#')
+                self.valid_selected_objects.append((allowed_objects[i], (tuple(int(c[i:i+2],16) for i in (0, 2, 4)), object_color), 1))
```
Comment (ian-r-rose) [inline automated_walk_bike_counter/gui/app.py]: Clever color parsing.

**Comment by ian-r-rose** [comment]
When I try to run this in a terminal without X11/GUI abilities, I get an error `cannot connect to X server`. I can start an X server and get past that point, but I think this should be able to run in the absence of any GUI/X server running. Somewhere tkinter is still initializing in console mode.

File: automated_walk_bike_counter/core/tracking/object_tracker.py [hunk 3]
```diff
@@ -569,27 +575,27 @@ def trackObjects(self, args):
...
                         videoWriter.write(self.current_frame.postprocessed_frame)
+                    if self.frame_listener:
                         self.update_frame_listener(
                             self.current_frame.postprocessed_frame,
                             self.currentFrameNumber,
                         )
-
-                    if self.input_camera_type == "webcam":
+                    if self.input_camera_type == "webcam" or not self.frame_listener:
                         cv2.imshow('', self.current_frame.postprocessed_frame)
```
Comment (ian-r-rose) [inline automated_walk_bike_counter/core/tracking/object_tracker.py]: It seems to be the calls to `cv2.imshow` that are failing when x11 is not running. Seems like we will need to add an additional check for CLI mode in those spots.

**Comment by ian-r-rose** [comment]
Any updates here @vahedi25411?

**Comment by ian-r-rose** [comment]
Can you respond to my review comments? There were several issues I flagged here.

**Comment by ian-r-rose** [comment]
@vahedi25411 are you not watching this repo? I'd like to have more communication on the status of the code go on here.

**Comment by vahedi25411** [comment]
Sorry, I had missed the messages of this repo before removing merge conflicts. I am answering your comments.

File: automated_walk_bike_counter/core/tracking/object_tracker.py (hunk 3 above)
Comment (vahedi25411) [reply automated_walk_bike_counter/core/tracking/object_tracker.py]: So in CLI mode, you don't want to see anything except the counting result, Right?

Comment (ian-r-rose) [reply automated_walk_bike_counter/core/tracking/object_tracker.py]: Yeah, I want to be able to run this in settings without any GUI available.

Comment (ian-r-rose) [reply automated_walk_bike_counter/core/tracking/object_tracker.py]: Eventually, we'd like to have some ability to plugin to export different count formats (such as SCAG ATDB), but I can work on that later.

Comment (vahedi25411) [reply automated_walk_bike_counter/core/tracking/object_tracker.py]: Got it. I am going to make changes in cli branch and then commit the changes.

Comment (ian-r-rose) [reply automated_walk_bike_counter/core/tracking/object_tracker.py]: Can you open a new PR with those changes, and run the `pre-commit` checks on the repo?

Comment (vahedi25411) [reply automated_walk_bike_counter/core/tracking/object_tracker.py]: Sure.

**Pull request merged**